File: app/src/main/java/com/botijasoftware/utils/Font.java
package com.botijasoftware.utils;


public class Font {

	public Font(int fontTexture, int size, Layout layout)
	{
	    fontTextureID = fontTexture;
		fontsize = size;
		mLayout = layout;
		font = null;
		atlas = null;
	}
	
	public void LoadContent(ResourceManager resources) {
		font = resources.loadTexture( fontTextureID );
		atlas = new SimpleTextureAtlas( font, 16, 16, 256 );
	}
	
	public Texture getTexture() {
		return font;
	}

    public Texture getTexture(int index) {
        return atlas.getAsTexture(index);
    }
	
	public TextureCoords getTextureCoords(int index) {
		return atlas.getTextureCoords(index);
	}
	
	public int getFontSize() {
		return fontsize;
	}
	
	private Texture font;
	private SimpleTextureAtlas atlas;
	private Layout mLayout;
	private int fontTextureID;
	private int fontsize;
}

File: app/src/main/java/com/botijasoftware/utils/Font.kt
package com.botijasoftware.utils


class Font(private val fontTextureID: Int, val fontSize: Int, private val mLayout: Layout) {
    lateinit var texture: Texture
    private lateinit var atlas: SimpleTextureAtlas

    init {
        //texture = null
        //atlas = null
    }

    fun LoadContent(resources: ResourceManager) {
        val texture = resources.loadTexture(fontTextureID)
        atlas = SimpleTextureAtlas(texture, 16, 16, 256)
    }

    fun getTexture(index: Int): Texture {
        return atlas.getAsTexture(index)
    }

    fun getTextureCoords(index: Int): TextureCoords {
        return atlas.getTextureCoords(index)

    }

}
